### PR TITLE
feat: IESG table column 'Pre pubreq'

### LINKED ISSUE #4577: Add Total Number of I-Ds to the I-D State Count table in the AD Dashboard
### Description

The I-D State Count table of the AD Dashboard (https://datatracker.ietf.org/doc/ad) currently lists the documents an AD is shepherding from PubReq to Approval.  To provide a sense of the load prior to PubReq, recommend adding a column as follows:

OLD I-D State Count
* Column 1 = AD Name
* Column = Pub Req

NEW I-D State Count
* Column 1 = AD Name
* Column 2 = Total I-Ds = a count of adopted I-Ds in the WGs overseen by the AD + any independent documents the AD is assigned to (in cases of AD sponsored documents) 
* Column 3 = Pub Req

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by holloway**
Screenshot from 2024-05-24 08-42-39

Fixes https://github.com/ietf-tools/datatracker/issues/4577

### CODE REVIEW

File: ietf/templates/doc/ad_list.html [hunk 1]
```diff
@@ -87,6 +103,37 @@ <h2 class="mt-5" id="{{ dt.type.0 }}">{{ dt.type.1 }} State Counts</h2>
     </script>
     <script src="{% static "ietf/js/highcharts.js" %}"></script>
     {{ data|json_script:"data" }}
+
+    <script>
+         Array.from(document.querySelectorAll("table"))
+            .filter(table => {
+                const hasDataSum = Boolean(table.querySelector("[data-sum]"))
+                return hasDataSum
```
Comment (NGPixel) [inline ietf/templates/doc/ad_list.html:111]: Nitpicking here: avoid creating a variable that's never used afterward. return the result directly.

File: ietf/templates/doc/ad_list.html [hunk 2]
```diff
@@ -87,6 +103,37 @@ <h2 class="mt-5" id="{{ dt.type.0 }}">{{ dt.type.1 }} State Counts</h2>
...
+            })
+            .forEach(table => {
+                const sums = Array.from(table.querySelectorAll("[data-sum]")).reduce((
+                    sums,
```
Comment (NGPixel) [inline ietf/templates/doc/ad_list.html:115]: There are 2 variables named "sums" in 2 different scopes, which can cause confusion when reading.

**Review by NGPixel — CHANGES_REQUESTED** [review]

**Review by jennifer-richards — COMMENTED** [review]
One inline comment. Also, as a general matter, it'd be nice to keep chipping away at moving javascript out of inline `<script>` tags and into their own js files. It's not exactly on-topic for this PR, though.

File: ietf/templates/doc/ad_list.html [hunk 3]
```diff
@@ -87,6 +103,37 @@ <h2 class="mt-5" id="{{ dt.type.0 }}">{{ dt.type.1 }} State Counts</h2>
...
+                    cell
+                ) => {                    
+                    const key = cell.dataset.sum
+                    const value = parseFloat(cell.textContent)
```
[the remark this answers is not in the document]
Comment (hollowaysmith) [reply ietf/templates/doc/ad_list.html:119]: So something like this?

```javascript
function safeParseFloat(text) {
        const trimNumber = text.trim()
        if(!trimNumber.match(/^[0-9.]+$/)) {
            console.warn(`Unable to parse "${trimNumber}" as a number.`)
            return Number.NaN
        }
        return parseFloat(text)
}
```

I think having tests for this is going to be hard with inline scripts but I guess I could put it into Playwright

[the remark this answers is not in the document]
Comment (holloway) [reply ietf/templates/doc/ad_list.html]: I'll try refactoring this JS into a separate script in the next PR.

File: playwright/tests-legacy/docs/ad.spec.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const { test, expect } = require('@playwright/test')
+const viewports = require('../../helpers/viewports')
+
+// ====================================================================
+// NOMCOM - EXPERTISE
```
Comment (NGPixel) [inline playwright/tests-legacy/docs/ad.spec.js:5]: Wrong title

File: ietf/templates/doc/ad_list.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (jennifer-richards) [reply ietf/templates/doc/ad_list.html]: That would do it, though we could also do it on the Django side just looking at the rendered template. That might be easier to turn into a test failure without having to add Playwright coverage or resort to `console.warn`.

File: ietf/templates/doc/ad_list.html (hunk 2 above)
Comment (holloway) [reply ietf/templates/doc/ad_list.html]: Fixed

File: ietf/templates/doc/ad_list.html (hunk 1 above)
Comment (holloway) [reply ietf/templates/doc/ad_list.html]: Fixed

**Review by NGPixel — APPROVED** [review]

**Pull request merged**